CMakeLists.txt

```diff
@@ -123,5 +125,38 @@ endif()
 ### Exports
 ###
 set(CMAKE_EXPORT_PACKAGE_REGISTRY ON)
-export(TARGETS mlspp tls_syntax hpke bytes mls_vectors third_party NAMESPACE MLSPP:: FILE MLSPPConfig.cmake)
+export(EXPORT mlspp-targets 
+       NAMESPACE MLSPP::
+       FILE ${CMAKE_CURRENT_BINARY_DIR}/mlspp-targets.cmake)
 export(PACKAGE MLSPP)
+
+configure_package_config_file(cmake/config.cmake.in
+  ${CMAKE_CURRENT_BINARY_DIR}/mlspp-config.cmake
+  INSTALL_DESTINATION ${CMAKE_INSTALL_DATADIR}/mlspp
+  NO_SET_AND_CHECK_MACRO)
+
+write_basic_package_version_file(
+  ${CMAKE_CURRENT_BINARY_DIR}/mlspp-config-version.cmake
+  VERSION ${PROJECT_VERSION}
+  COMPATIBILITY SameMajorVersion)
+
+###
+### Install
+###
+
+install(TARGETS ${LIB_NAME} EXPORT mlspp-targets)
```

Comment:
Could we avoid duplication in other CMake files by putting the other targets under `TARGETS` here?

Reply:
I believe that CMAKE configurations make more sense, the closer they are to the code.  That does cause some duplication, but it prevents any one small change from having bigger consequences.